conjure-java-core/src/test/java/com/palantir/conjure/java/types/UnionTests.java

```diff
@@ -207,6 +213,53 @@ void unionWithDefensiveCollectionsIsImmutable_map() {
         assertThatExceptionOfType(UnsupportedOperationException.class).isThrownBy(() -> internal.put("test", "update"));
     }
 
+    @Test
+    void sealedUnionCanBeConsumedWithSwitchStatement() {
+        String expected = "foo";
+
+        String s =
+                switch (SimpleUnion.foo(expected).throwOnUnknown()) {
```

Comment:
hm why do we need the `throwOnUnknown` in `SimpleUnion`? We're constructing the object so I think this is unneeded (unless you were planning on creating some sort of visitor for back compat?)